ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/InMemory.hs

```diff
@@ -164,9 +155,53 @@ type ModernLedgerDB l = DbChangelog l
 -- blocks. For example, if we are on line (*), and roll back 6 blocks, we get
 --
 -- > L3 |> []
+type LegacyLedgerDB l = AnchoredSeq
+                               (WithOrigin SlotNo)
+                               (Checkpoint (l ValuesMK))
+                               (Checkpoint (l ValuesMK))
+
+-- | Type alias for the new-style database
+--
+-- This type is internally a DbChangelog which as an example, with @k = 3@ could look like:
```

Comment:
I wonder if this comment belongs in the definition of `DbChangelog`.

Reply:
DbChangelog to me is just a data structure. It is this module the one that defines how it works and is used.